R/check_spatvector.R

```diff
@@ -0,0 +1,145 @@
+#' Check SpatVector Properties
+#'
+#' This function checks if a given object is a valid SpatVector polygon with
+#' appropriate coordinate system and size constraints, and validates its overlap
+#' with known country boundaries.
+#'
+#' @param x Object to check.
+#' @param check_size Logical. Whether to check if the area is within reasonable bounds.
+#' Default is TRUE.
+#'
+#' @return The input SpatVector, possibly reprojected to EPSG:4326.
+#'
+#' @details
+#' The function checks:
+#' - If input is a SpatVector polygon
+#' - If coordinate system is properly defined
+#' - If polygons are valid and non-empty
+#' - If the area overlaps with known country boundaries
+#' - Optionally checks if area is within reasonable bounds
+#' (larger than 10,000 ha but smaller than approximately Brazil)
+#'
+#' @import terra
+#'
+#' @references
+#' Zillah Calle (2023)
+#' Jonas van Duijvenbode (2023)
+#'
+#' @examples
+#' \dontrun{
+#' shape <- vect("path/to/shapefile.shp")
+#' validated_shape <- check_spatvector(shape)
+#' }
+#'
+#' @export
+check_spatvector <- function(x, check_size = TRUE) {
+  x <- check_basic_properties(x)
+  x <- check_coordinate_system(x)
+  x <- check_country_overlap(x)
+  if (check_size) {
+    check_area_bounds(x)
+  }
+  return(x)
+}
+
+#' Check basic SpatVector properties
+#' @noRd
+check_basic_properties <- function(x) {
+  if (!inherits(x, "SpatVector")) {
+    stop(paste0("Input must be a SpatVector object, received ", class(x)))
+  }
+  if (terra::geomtype(x) != "polygons") {
+    stop(paste("Input must be a polygon SpatVector, received", terra::geomtype(x)))
+  }
+  if (terra::nrow(x) == 0) {
+    stop("Input SpatVector contains no polygons")
+  }
+  return(x)
+}
+
+#' Check and fix coordinate system
+#' @noRd
+check_coordinate_system <- function(x) {
+  if (is.na(terra::crs(x))) {
+    x <- handle_missing_crs(x)
+  } else if (!is_wgs84(terra::crs(x))) {
+    ff_cat("Coordinate system is not EPSG:4326. Converting to WGS84.", color = "yellow")
```

Comment:
This is probably just me not knowing what these abbreviations mean. But it says: "Converting to WGS84", and then the following line projects to "ESPG:4326". To me that's confusing, but if it's correct than ignore this comment.

Reply:
yeah they are synonyms.